jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper;
+
+import io.opentelemetry.contrib.jmxscraper.client.JmxRemoteClient;
+import io.opentelemetry.contrib.jmxscraper.config.ConfigurationException;
+import io.opentelemetry.contrib.jmxscraper.config.JmxScraperConfig;
+import io.opentelemetry.contrib.jmxscraper.config.JmxScraperConfigFactory;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.logging.Logger;
+import javax.management.MBeanServerConnection;
+import javax.management.remote.JMXConnector;
+
+public class JmxScraper {
+  private static final Logger logger = Logger.getLogger(JmxScraper.class.getName());
+
+  private final JmxRemoteClient client;
+
+  // TODO depend on instrumentation 2.9.0 snapshot
+  // private final JmxMetricInsight service;
+
+  /**
+   * Main method to create and run a {@link JmxScraper} instance.
+   *
+   * @param args - must be of the form "-config {jmx_config_path,'-'}"
+   */
+  @SuppressWarnings({"SystemOut", "SystemExitOutsideMain"})
+  public static void main(String[] args) {
+    JmxScraperConfig config;
+    JmxScraper jmxScraper = null;
+    try {
+      JmxScraperConfigFactory factory = new JmxScraperConfigFactory();
+      config = JmxScraper.createConfigFromArgs(Arrays.asList(args), factory);
+      jmxScraper = new JmxScraper(config);
+
+    } catch (ArgumentsParsingException e) {
+      System.err.println(
+          "Usage: java -jar <path_to_jmxscraper.jar> "
+              + "-config <path_to_config.properties or - for stdin>");
+      System.exit(1);
+    } catch (ConfigurationException e) {
+      System.err.println(e.getMessage());
+      System.exit(1);
+    }
+
+    try {
+      Objects.requireNonNull(jmxScraper).start();
```

Comment:
requireNonNull call is redundant and can be removed.

Reply:
I've removed the `requireNonNull`.

For the shutdown hook, I don't think it's needed anymore as the polling and sending is now managed by the SDK (but I might be wrong here, and we will eventually discover this with end-to-end tests with short-lived JVMs).

From what I understand, the task scheduling in `io.opentelemetry.instrumentation.jmx.engine.BeanFinder#discoverBeans` handles the discovery and registration of JMX beans metrics, and the `io.opentelemetry.sdk.metrics.export.PeriodicMetricReader` in the SDK deals with exporing the metrics at a regular interval so this should be already covered thanks to SDK auto-configuration in `io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder` that registers `io.opentelemetry.sdk.OpenTelemetrySdk#close` to be called as shutdown hook.

Reply:
In fact the `requirenonNull` is needed because nullaway check fails build otherwise.